### PR TITLE
Remove String#freeze calls

### PR DESCRIPTION
**Pull request by mperham**
Explicit freezing was required pre-Ruby 2.3.  With 2.3, `frozen_string_literal: true` is available to do the same thing without uglifying the code.

Caveat upgrader: this change will lead to increased memory usage and slower performance on Ruby 2.2.

### CODE REVIEW

File: lib/sidekiq.rb [hunk 1]
```diff
@@ -48,7 +47,7 @@ module Sidekiq
     "connected_clients" => "9999",
     "used_memory_human" => "9P",
     "used_memory_peak_human" => "9P"
-  }.freeze
```
Comment (SamSaffron) [inline lib/sidekiq.rb]: note this is a behavior change cause Hash is not a string literal so the hash will no longer be frozen

Comment (mperham) [reply lib/sidekiq.rb]: In this case, it's not a concern.

**Comment by krzysiek1507** [comment]
Didn't you miss it in those files?
```
lib/sidekiq/delay.rb
lib/sidekiq/job_logger.rb
lib/sidekiq/job_retry.rb
lib/sidekiq/middleware/server/active_record.rb
lib/sidekiq/rails.rb
```

**Comment by mperham** [comment]
thanks for the check, fixed!

**Comment by krzysiek1507** [comment]
You're welcome!

**Pull request merged**